### PR TITLE
Added input validation for new metadata field submissions & fixed some other small bugs

### PR DESCRIPTION
**Pull request by alexandrevryghem**
## References
* Fixes DSpace/DSpace#8753
* Fixes DSpace/DSpace#8754
* Related DSpace/DSpace#8758

## Description
In this PR I added the input validation to the Metadata Registry forms (and fixed some other small issues related to those forms) and I also set the max suggestion dropdown length back to 10. I also made the frontend changes to prevent users from editing the metadata schema Name, Element or qualifier by disabling those fields during the update.

## Instructions for Reviewers
List of changes in this PR:
* `MetadataFieldFormComponent` & `MetadataSchemaFormComponent`
  * Added validation for the fields to check that they don't contain any dots
  * Removed error from first field after successfully submitting the form or unselecting the field you are currently updating
  * For `MetadataSchemaFormComponent` only: cleared out the name field after successfully submitting the form
  * Disabled the schema Name, Element and Qualifier fields during the update process to prevent users from changing those values
* `MetadataFieldSelectorComponent`: The default page size on REST for all endpoints is 20 elements previously only 10 elements were shown because the pageable parameter on the rest side was not working. Now that it has been fixed (see related PR), I had to add that limit back so that only 10 suggestions are shown.

**Guidance for how to test & review this PR:**
- Go to `/admin/registries/metadata`
  - Try adding a valid new schema and check that both fields are cleared out & that the first field does not show an error
  - Fill in the Namespace input field with something containing a dot, check that an appropriate message is shown and that you can't submit it
- Go to `/admin/registries/metadata/dc`
  - Try adding a valid field and check that both fields are cleared out & that the first field does not show an error
  - Fill in the Element & Qualifier input fields with something containing a dot, check that an appropriate message is shown and that you can't submit it

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/assets/i18n/en.json5 [hunk 1]
```diff
@@ -1620,6 +1620,12 @@
 
   "error.validation.groupExists": "This group already exists",
 
+  "error.validation.metadata.namespace.invalid-pattern": "This field cannot contain dots, please use the Element & Qualifier fields instead",
+
+  "error.validation.metadata.element.invalid-pattern": "This field cannot contain dots, please use the Qualifier field instead",
+
+  "error.validation.metadata.qualifier.invalid-pattern": "This field cannot contain dots",
+
```
Comment (abollini) [inline src/assets/i18n/en.json5:1628]: Please update it according to the final excluded chars, This field cannot contain inner spaces, dots or underscore _, please ...

**Review by abollini — COMMENTED** [review]
code looks good to me at a first glace I will make a try once the REST PR is finalized

File: src/assets/i18n/en.json5 [hunk 2]
```diff
@@ -1620,6 +1620,12 @@
 
   "error.validation.groupExists": "This group already exists",
 
+  "error.validation.metadata.namespace.invalid-pattern": "This field cannot contain dots, please use the Element & Qualifier fields instead",
+
+  "error.validation.metadata.element.invalid-pattern": "This field cannot contain dots, please use the Qualifier field instead",
+
+  "error.validation.metadata.qualifier.invalid-pattern": "This field cannot contain dots",
```
Comment (tdonohue) [inline src/assets/i18n/en.json5:1627]: If you agree with my feedback above, then this should be updated to say: "This field cannot contain dots, commas or spaces."

File: src/app/admin/admin-registries/metadata-schema/metadata-field-form/metadata-field-form.component.ts [hunk 3]
```diff
@@ -98,25 +98,35 @@ export class MetadataFieldFormComponent implements OnInit, OnDestroy {
...
       this.translateService.get(`${this.messagePrefix}.scopenote`)
-    ).subscribe(([element, qualifier, scopenote]) => {
+    ]).subscribe(([element, qualifier, scopenote]) => {
       this.element = new DynamicInputModel({
         id: 'element',
         label: element,
         name: 'element',
         validators: {
           required: null,
+          pattern: '^[^.]*$',
```
Comment (tdonohue) [inline src/app/admin/admin-registries/metadata-schema/metadata-field-form/metadata-field-form.component.ts:112]: I think we should block commas and spaces here as well. (But should allow underscores)  So, this should be:  `'^[^. ,]*$',`

I tested and was able to add a metadata field with a space in the element and qualifier. That seems wrong.

File: src/app/admin/admin-registries/metadata-schema/metadata-field-form/metadata-field-form.component.ts [hunk 4]
```diff
@@ -98,25 +98,35 @@ export class MetadataFieldFormComponent implements OnInit, OnDestroy {
...
+        errorMessages: {
+          pattern: 'error.validation.metadata.element.invalid-pattern',
+        },
       });
       this.qualifier = new DynamicInputModel({
         id: 'qualifier',
         label: qualifier,
         name: 'qualifier',
+        validators: {
+          pattern: '^[^.]*$',
```
Comment (tdonohue) [inline src/app/admin/admin-registries/metadata-schema/metadata-field-form/metadata-field-form.component.ts:124]: Same as above, should block commas and spaces as well.  (But should allow underscores)

**Review by tdonohue — CHANGES_REQUESTED** [review]
@alexandrevryghem : Thanks for this!  Overall this looks good and it works well.  It definitely fixes the reported issues.  However, in testing, I located a few minor issues which I've noted inline below.  I think these are all very quick to fix though, and I'd be +1 after these changes are made.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @alexandrevryghem !  Re-tested today with the backend PR and this improved validation is all working well.

**Pull request merged**